File: blog-server/src/module/admin/routingConfigure/routingConfigure.controller.ts
import {Controller, Post, Delete, Body, HttpStatus} from '@nestjs/common'
import { RoutingConfigureService } from './routingConfigure.service'
import {
    RoutingConfigureFindParam,
    RoutingConfigureInsertParam,
    RoutingConfigureUpdateParam
} from './model/routingConfigure.param'
import {ApiTags, ApiResponse, ApiBody} from '@nestjs/swagger';
import {IdParam} from "../../../model/BaseParam";
import { RoutingConfigureFindRsutl} from "./model/RoutingConfigure.Rsult";


@ApiTags('routingConfigure')  // 用于给路由分组
@Controller('api/routingConfigure')
export class RoutingConfigureController {
    constructor(
        private routingService: RoutingConfigureService
    ) {}

    @ApiBody({ type: RoutingConfigureInsertParam })  // 描述请求体
    @ApiResponse({status: HttpStatus.CREATED, description: '成功'})
    @Post('insert')
    insert (@Body() insertBody: RoutingConfigureInsertParam ): Promise<void> {
        return this.routingService.insert(insertBody)
    }

    @ApiBody({ type: RoutingConfigureUpdateParam })  // 描述请求体
    @ApiResponse({status: HttpStatus.CREATED, description: '成功'})
    @Post('update')
    update (@Body() updateBody: RoutingConfigureUpdateParam ): Promise<void> {
        return this.routingService.update(updateBody)
    }

    @ApiBody({ type: RoutingConfigureFindParam })  // 描述请求体
    @ApiResponse({status: HttpStatus.CREATED, type: Array<RoutingConfigureFindRsutl>, description: '成功'})
    @Post('list')
    list (@Body() listBody: RoutingConfigureFindParam ): Promise<Array<RoutingConfigureFindRsutl>> {
        return this.routingService.list(listBody)
    }

    @ApiBody({ type: IdParam })  // 描述请求体
    @ApiResponse({status: HttpStatus.CREATED, type: Array<RoutingConfigureFindRsutl>, description: '成功'})
    @Post('getPowerRuoter')
    getPowerRouter (@Body() powerBody: IdParam ): Promise<Array<RoutingConfigureFindRsutl>> {
        return this.routingService.getPowerRouter(powerBody)
    }

    @ApiBody({ type: RoutingConfigureInsertParam })  // 描述请求体
    @ApiResponse({status: HttpStatus.CREATED, description: '成功'})
    @Delete('delete')
    delete (@Body() deleteBody: IdParam ): Promise<void> {
        return this.routingService.delete(deleteBody)
    }
}

File: BlogServer/Blog.Web/Controllers/Api/RoutingConfigureController.cs
using Blog.Model.Param;
using Blog.Model.Rsult;
using Blog.Service.Api.Interface;
using Blog.Utils;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;

namespace Blog.Web.Controllers.Api
{
    [Route("api/routingConfigure")]
    [ApiController]
    [Authorize]
    public class RoutingConfigureController : ControllerBase
    {
        public RoutingConfigureController(IRoutingConfigureService routingConfigureService)
        {
            RoutingConfigureService = routingConfigureService;
        }
        public IRoutingConfigureService RoutingConfigureService { get; }

        [HttpPost("insert")]
        public async Task<ResultStruct> Insert([FromBody] RoutingConfigureInsertParam param)
        {
            return await ResultFun.AsyncReturn(param, RoutingConfigureService.Insert);
        }

        [HttpPost("list")]
        public async Task<ResultStruct<List<RoutingConfigureFindRsult>>> List([FromBody] RoutingConfigureFindParam param)
        {
            return await ResultFun.AsyncReturn(param, RoutingConfigureService.List);
        }

        [HttpPost("update")]
        public async Task<ResultStruct> Update([FromBody] RoutingConfigureUpdateParam param)
        {
            return await ResultFun.AsyncReturn(param, RoutingConfigureService.Update);
        }

        [HttpPost("getPowerRuoter")]
        public async Task<ResultStruct<List<RoutingConfigureFindRsult>>> GetPowerRuoter([FromBody] IDParam param)
        {
            return await ResultFun.AsyncReturn(param, RoutingConfigureService.GetPowerRuoter);
        }

        [HttpDelete("delete")]
        public async Task<ResultStruct> Delete([FromBody] IDParam param)
        {
            return await ResultFun.AsyncReturn(param, RoutingConfigureService.Delete);
        }
    }
}
